bigchaindb/common/transaction.py

```diff
@@ -1194,6 +1197,13 @@ def from_dict(cls, tx_body):
             Returns:
                 :class:`~bigchaindb.common.transaction.Transaction`
         """
+
+        # Attempt full validation of tx body, fail quickly if possible
+        try:
+            jsonschema.validate(tx_body, TX_JSON_SCHEMA)
+        except jsonschema.ValidationError as e:
+            raise ValidationError(str(e))
```

Comment:
Yep, agreed. Question though, should verification of transaction ID remain inside `from_dict`?